lightning-invoice/src/payment.rs

```diff
@@ -744,77 +752,105 @@ fn has_expired(route_params: &RouteParameters) -> bool {
 	} else { false }
 }
 
-impl<P: Deref, R: ScoringRouter, L: Deref, E: EventHandler, T: Time> EventHandler for InvoicePayerUsingTime<P, R, L, E, T>
+impl<P: Deref, R: ScoringRouter, L: Deref, E: sealed::BaseEventHandler, T: Time> InvoicePayerUsingTime<P, R, L, E, T>
 where
 	P::Target: Payer,
 	L::Target: Logger,
 {
-	fn handle_event(&self, event: &Event) {
+	fn handle_event_internal(&self, event: Event) -> Option<Event> {
```

Comment:
I don't have a strong opinion here, but since this is no longer an `EventHandler` implementation, you can still take the `Event` by reference and return a `bool` instead of moving in and back out. But maybe the new interface is clearer.